src/components/DateTime/DateTime.jsx

```diff
@@ -39,3 +43,53 @@ export function DateTime(props) {
 		</span>
 	);
 }
+
+function validateDateTime(value) {
+	// Return 'Invalid Date' if the input is null or undefined
+	if (value == null) {
+		return 'Invalid Date';
+	}
+
+	// Handle BigInt values explicitly
+	if (typeof value === 'bigint') {
+		// SP-Lang datetime format is represented as BigInt
+		return splDatetimeToIso(value);
+	}
+
+	// Handle string values
+	if (typeof value === 'string') {
+		return new Date(value);
+	}
+
+	// Handle number values
+	if (typeof value === 'number') {
+		// Reject infinite, NaN, or negative numbers
+		if (!Number.isFinite(value) || value < 0) {
+			return 'Invalid Date';
+		}
+
+		// Consider numbers ≥ 1e17 as SP-Lang datetime format
+		if (value >= 1e17) {
+			// Convert number to BigInt and parse as SP-Lang datetime
+			return splDatetimeToIso(BigInt(value));
```

Comment:
If we convert to BigInt here, we already lost a precision - so this is incorrect.